tests/load/test_dummy_client.py

```diff
@@ -107,15 +107,24 @@ def test_unsupported_write_disposition() -> None:
 
 
 def test_big_loadpackages() -> None:
-    import time
+    """
+    This test guards against changes in the load that exponentially makes the loads slower
+    """
 
-    start_time = time.time()
     load = setup_loader()
     load_id, schema = prepare_load_package(load.load_storage, SMALL_FILES, jobs_per_case=500)
-    print("start" + str(time.time() - start_time))
+    start_time = time()
     with ThreadPoolExecutor(max_workers=20) as pool:
         load.run(pool)
-    print("done" + str(time.time() - start_time))
+    duration = float(time() - start_time)
+
+    # sanity check
+    assert duration > 5
+
+    # we want 1000 empty processed jobs to need less than 15 seconds total (locally it runs in 10)
```

Comment:
I did the same test on devel and it is way slower there, I'd need to dig further into it why that is the case, maybe the long sleep and the restarting of the load, or maybe something else idk.

Reply:
I can go back to 1 second, no problem (and maybe modify it for the speed test). If you look at "start_new_jobs", I made a change in the last round so that it checks how many slots are available before actually querying any files on the disc, so if all slots are full, it exits early and nothing is done. I'd say you can tell that it is not starving in my speed test where many files are present on the disc.

Reply:
The number for running empty jobs in this test are:
1000 jobs = 4s
2000 jobs = 12s
10k jobs = 372s
So it does go up quite a bit, but definately not from listing all files on every loop (as it does not do this). What I would do: Merge this like it is now, it already is faster than the old implementation, and in a followup ticket with plenty tests we cache the parsed jobs in some in memory structure that gets refreshed periodically or when the new jobs in this structure fall below the count of open slots.